community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/UpgradableDatabase.java

```diff
@@ -84,6 +87,13 @@ public RecordFormats checkUpgradeable( File storeDirectory )
             return format;
         }
 
+        // If we are trying to open an enterprise store when configured to use community format, then inform the user
+        // of the config setting to change since downgrades aren't possible but the store can still be opened.
+        if ( result.actualVersion.startsWith( "vE" ) && !format.storeVersion().startsWith( "vE" ) )
```

Comment:
`result` IS `fromFormat`. Problem is that you can't instantiate the `fromFormat` object from result because the enterprise format is not a known format in this case, resulting in `IllegalArgumentException( "Unknown store version '" + storeVersion + "'" )` (see `RecordFormatSelector.selectForVersion( result.actualVersion )`).

I can move the if-clause above into `RecordFormatSelector` if you prefer, e.g. something like:

In `RecordFormatSelector`:

```java
    /**
     * Select record formats for provided store version.
     *
     * @param storeVersion store version to find format for
     * @return record formats
     * @throws IllegalArgumentException if format for specified store version not found
     */
    public static RecordFormats selectForVersion( String storeVersion )
    {
        Iterable<RecordFormats> currentFormats =
                map( RecordFormats.Factory::newInstance, Service.load( RecordFormats.Factory.class ) );
        for ( RecordFormats format : concat( KNOWN_FORMATS, currentFormats ) )
        {
            if ( format.storeVersion().equals( storeVersion ) )
            {
                return format;
            }
        }
        // If we are trying to open an enterprise store but this is not a known format, 
        // e.g. we are running community, then inform the user
        // of the config setting to change since downgrades aren't possible but the store can still be opened.
        if ( storeVersion.startsWith( "vE" ) )
        {
            throw new StoreUpgrader.UnexpectedUpgradingStoreFormatException();
        }
        throw new IllegalArgumentException( "Unknown store version '" + storeVersion + "'" );
    }
```

What do you think @lutovich ?